controller/instance_manager_controller.go

```diff
@@ -1662,13 +1662,11 @@ func (imc *InstanceManagerController) createInstanceManagerPodSpec(im *longhorn.
 
 func (imc *InstanceManagerController) deleteOrphans(im *longhorn.InstanceManager, isInstanceManagerTerminating bool) error {
 	autoDeletionTypes, err := imc.ds.GetSettingOrphanResourceAutoDeletion()
-	var autoDeleteEngine = false
-	var autoDeleteReplica = false
+	var autoDeleteEnabled = false
 	if err != nil {
 		imc.logger.WithError(err).Warnf("Failed to fetch orphan auto deletion setting, disabled by default")
 	} else {
-		autoDeleteEngine = autoDeletionTypes[types.OrphanResourceTypeEngineInstance]
-		autoDeleteReplica = autoDeletionTypes[types.OrphanResourceTypeReplicaInstance]
+		autoDeleteEnabled = autoDeletionTypes[types.OrphanResourceTypeInstance]
```

Comment:
Would it be better to return directly if auto-delete is disabled?

Reply:
(see the reason at https://github.com/longhorn/longhorn-manager/pull/3774#discussion_r2092188262)